Example/Tests/Classes/LPRequestFactoryTest.m

```diff
@@ -0,0 +1,442 @@
+NSString *LP_API_METHOD_MARK_INBOX_MESSAGE_AS_READ = @"markNewsfeedMessageAsRead";
+NSString *LP_API_METHOD_DELETE_INBOX_MESSAGE = @"deleteNewsfeedMessage";
+
+@interface LPRequestFactory(UnitTest)
+
+@property (nonatomic, strong) LPFeatureFlagManager *featureFlagManager;
+
+- (id<LPRequesting>)createGetForApiMethod:(NSString *)apiMethod params:(NSDictionary *)params;
+- (id<LPRequesting>)createPostForApiMethod:(NSString *)apiMethod params:(NSDictionary *)params;
+- (BOOL)shouldReturnLPRequestClass;
+
+@end
+
+@interface LPRequestFactoryTest : XCTestCase
+
+@end
+
+@implementation LPRequestFactoryTest
+
+- (void)setUp {
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testCreateGetForApiMethodLPRequest {
+    LPFeatureFlagManager *featureFlagManager = [[LPFeatureFlagManager alloc] init];
+    LPRequestFactory *reqFactory = [[LPRequestFactory alloc]
+                                    initWithFeatureFlagManager:featureFlagManager];
+    id LPRequestMock = OCMClassMock([LPRequest class]);
+    reqFactory.featureFlagManager = OCMClassMock([LPFeatureFlagManager class]);
+    OCMStub([reqFactory.featureFlagManager isFeatureFlagEnabled:LP_FEATURE_FLAG_REQUEST_REFACTOR]).andReturn(true);
+    NSMutableDictionary *params = [[NSMutableDictionary alloc] init];
+    params[@"key"] = @"value";
+    NSString *apiMethod = @"ApiMethod";
+    
+    [reqFactory createGetForApiMethod:apiMethod params:params];
+    
+    OCMVerify([LPRequestMock get:apiMethod params:params]);
+}
+
+- (void)testCreateGetForApiMethodLeanplumRequest {
+    LPFeatureFlagManager *featureFlagManager = [[LPFeatureFlagManager alloc] init];
+    LPRequestFactory *reqFactory = [[LPRequestFactory alloc]
+                                    initWithFeatureFlagManager:featureFlagManager];
+    id LeanplumRequestMock = OCMClassMock([LeanplumRequest class]);
+    
+    NSMutableDictionary *params = [[NSMutableDictionary alloc] init];
+    params[@"key"] = @"value";
+    NSString *apiMethod = @"ApiMethod";
+    
+    [reqFactory createGetForApiMethod:apiMethod params:params];
+    
+    OCMVerify([LeanplumRequestMock get:apiMethod params:params]);
+}
+
+- (void)testCreatePostForApiMethodLPRequest {
+    LPFeatureFlagManager *featureFlagManager = [[LPFeatureFlagManager alloc] init];
+    LPRequestFactory *reqFactory = [[LPRequestFactory alloc]
+                                    initWithFeatureFlagManager:featureFlagManager];
+    id LPRequestMock = OCMClassMock([LPRequest class]);
+    reqFactory.featureFlagManager = OCMClassMock([LPFeatureFlagManager class]);
+    OCMStub([reqFactory.featureFlagManager isFeatureFlagEnabled:LP_FEATURE_FLAG_REQUEST_REFACTOR]).andReturn(true);
+    
+    NSMutableDictionary *params = [[NSMutableDictionary alloc] init];
+    params[@"key"] = @"value";
```

Comment:
Nit: Passing a param here is unnecessary and would take in extra cycles in testing. This is totally a nit and you can ignore it to merge.

In case you make that change, it'd be great to enhance the code by adding nullability to `createPost` and `createGet`. They can both take in in `nil` parameters. Could we mark those methods in `LPRequest` with `nullable`